cabal-install/tests/IntegrationTests2.hs

```diff
@@ -869,15 +869,6 @@ testTargetProblemsRun config reportSubCase = do
       [ ( CmdRun.TargetProblemNoTargets, mkTargetPackage "p-0.1" )
       ]
 
-    reportSubCase "test-only"
```

Comment:
I'm not super happy of simply removing this test. 

It's invalid with this change, as https://github.com/haskell/cabal/blob/fcabd87406effc0c265312eb38509804682e4623/cabal-install/tests/IntegrationTests2/targets/test-only/p.cabal has only `test-suite` stanza, but now *has an executable*.

Let's change that package to be `lib-only` (you cannot run that).

What is pity, we don't have any positive tests for `CmdRun` in `IntegrationsTests2.hs`. We really should add them, so the things working now continue working in the future.